src/shared/components/TitleArea/TitleArea.tsx

```diff
@@ -1,39 +1,80 @@
-import React from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 
-import { Container, HelperText, StyledInput } from './TitleArea.style'
+import { CharactersCounter, Container, MinMaxChars, StyledInput, TitleAreaInfo } from './TitleArea.style'
+
+import { Text } from '../Text'
 
 export type TitleAreaProps = {
   name?: string
   value?: string
-  helperText?: string
+  min?: number
+  max?: number
   placeholder?: string
-  error?: boolean
-  warning?: boolean
-  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
-  onBlur?: (event: React.ChangeEvent<HTMLInputElement>) => void
+  onChange?: (event: React.ChangeEvent<HTMLTextAreaElement>) => void
+  onBlur?: (event: React.ChangeEvent<HTMLTextAreaElement>) => void
   className?: string
 }
 
-export const TitleArea = React.forwardRef<HTMLInputElement, TitleAreaProps>(
-  ({ name, value, helperText, placeholder, error, warning, onChange, onBlur, className }, ref) => {
-    const controlled = onChange?.name === 'onChange'
+export const TitleArea = React.forwardRef<HTMLTextAreaElement, TitleAreaProps>(
+  ({ name, value, placeholder = 'Enter text here', onChange, onBlur, className, max = 60, min = 5 }, ref) => {
+    const [charactersCount, setCharactersCount] = useState(0)
+    const [touched, setTouched] = useState(false)
+    const invalidInput = charactersCount < min || charactersCount > max
+    const containerRef = useRef<HTMLDivElement>(null)
+
+    const handleFocus = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      setCharactersCount(e.target.value.length)
+      setTouched(true)
+    }
+
+    useEffect(() => {
+      const textarea = containerRef.current?.querySelector('textarea')
+      if (!textarea) {
+        return
+      }
+      textarea.style.height = 'initial'
+      const scrollHeight = textarea.scrollHeight
+      textarea.style.height = scrollHeight + 'px'
+    })
```

Comment:
I really don't like it but I need this `useEffect`. The component needs to update its height somehow when you switch video tabs.
To be honest, I ran out of ideas on how to do this better.

Reply:
I don't understand why we need this in the first place, could you describe what problem this is solving?

Reply:
Basically, we need these lines to solve this issue: 
> The component docs specify that this component should actually wrap to two lines if the text doesn't fit. This makes sense for longer titles

When you typing in `textarea` and if the text starts going to the next line,`textarea` doesn't change its size. It just adds a scrollbar on the right. So, to solve that, my first attempt was to update the height of this textarea on every change - basically, I just put analogical code inside `handleOnChange`, but it wasn't enough. If you open for example one tab with a video that has a title that expands to two lines and then you change to another video that has a very short title, the TitleArea will not update itself and you'll have two-row input. I hope it makes sense. 

https://user-images.githubusercontent.com/51168865/132349073-02d8d427-2ad9-4587-96dd-89c51cf4528c.mov

Reply:
Ah okay, I forgot text areas don't expand by themselves. Have you seen this: https://css-tricks.com/the-cleanest-trick-for-autogrowing-textareas/ ? Looks interesting and doesn't need any javascript

Reply:
Woah! that looks cool. I don't know how I missed that! Will try to implement this.

Reply:
> Ah okay, I forgot text areas don't expand by themselves. Have you seen this: https://css-tricks.com/the-cleanest-trick-for-autogrowing-textareas/ ? Looks interesting and doesn't need any javascript

Unfortunately, this won't fix all the issues. When I'm switching tabs I still don't know what input is inside the textarea, because we're using uncontrolled components here, so basically, I don't know what the `value` is. If I don't know the `value` I cannot set `data-replicated-value` , this method is using javascript  after all - If you look at HTML in the link you'll see: 
```tsx
  <div class="grow-wrap">
    <textarea name="text" id="text" onInput="this.parentNode.dataset.replicatedValue = this.value"></textarea>
  </div>
```

I'm wondering. Can I just pass `value` like this?  

```tsx
// EditVideoForm line 438
            <StyledTitleArea
              {...register(
                'title',
                textFieldValidation({ name: 'Video Title', minLength: 3, maxLength: 40, required: true })
              )}
              value={getValues('title')} // can I just add this?
              min={3}
              max={40}
              placeholder="Video title"
            />
```
I guess it will rerender TitleArea on every input? Maybe with delay, because it looks like `getValues` is debounced(?)